pkg/config/draftconfig.go

```diff
@@ -125,6 +127,61 @@ func (d *DraftConfig) ApplyDefaultVariables() error {
 	return nil
 }
 
+func (d *DraftConfig) ApplyDefaultVariablesForVersion(version string) error {
+	v, err := semver.Parse(version)
+	if err != nil {
+		return fmt.Errorf("invalid version: %w", err)
+	}
+
+	expectedConfigVersionRange, err := semver.ParseRange(d.Versions)
+	if err != nil {
+		return fmt.Errorf("invalid config version range: %w", err)
+	}
+
+	if !expectedConfigVersionRange(v) {
+		return fmt.Errorf("version %s is outside of config version range %s", version, d.Versions)
+	}
+
+	for _, variable := range d.Variables {
+		if variable.Value == "" {
+			expectedRange, err := semver.ParseRange(variable.Versions)
+			if err != nil {
+				return fmt.Errorf("invalid variable versions: %w", err)
+			}
+
+			if !expectedRange(v) {
+				log.Infof("Variable %s versions %s is outside input version %s, skipping", variable.Name, variable.Versions, version)
+				continue
+			}
+
+			if variable.Default.ReferenceVar != "" {
+				referenceVar, err := d.GetVariable(variable.Default.ReferenceVar)
+				if err != nil {
+					return fmt.Errorf("apply default variables: %w", err)
+				}
+
+				defaultVal, err := d.recurseReferenceVars(referenceVar, referenceVar, true)
```

Comment:
maybe should've been caught earlier but this method has the potential to infinitely recurse if the cycle has length >1 right? Like if x references y which references z which references x?

Reply:
iterative approach with like a hash set or map or something might catch this iirc

Reply:
I see the implementation in the validation/test does account for this so hypothetically I guess there's no way that one of our templates could have this issue, so I guess this is fine

Reply:
the validation test code would actually miss the case where the cycle isnt the initial variable so I fixed it!